source/adapters/level_zero/command_buffer.cpp

```diff
@@ -170,6 +170,95 @@ ur_result_t getEventsFromSyncPoints(
   return UR_RESULT_SUCCESS;
 }
 
+/**
+ * If necessary, it creates a signal event and appends it to the previous
+ * command list (copy or compute), to indicate when it's finished executing.
+ * @param[in] CommandBuffer The CommandBuffer where the command is appended.
+ * @param[in] ZeCommandList the CommandList that's currently in use.
+ * @param[in] IsCopy A boolean indicating if the current command uses copy
+ * engine.
+ * @param[out] WaitEventList The list of event for the future command list to
+ * wait on before execution.
+ * @return UR_RESULT_SUCCESS or an error code on failure
+ */
+ur_result_t createSyncPointBetweenCopyAndCompute(
+    ur_exp_command_buffer_handle_t CommandBuffer,
+    ze_command_list_handle_t ZeCommandList, bool IsCopy,
+    std::vector<ze_event_handle_t> &WaitEventList) {
+
+  // Skip in cases when we're not using Copy engine or it's not in-order path.
+  if (!CommandBuffer->IsInOrderCmdList || !CommandBuffer->ZeCopyCommandList ||
+      (IsCopy && ZeCommandList != CommandBuffer->ZeCopyCommandList)) {
```

Comment:
> (IsCopy && ZeCommandList != CommandBuffer->ZeCopyCommandList)

Do we need a `IsCopy` function parameter? or could we deduce `IsCopy` inside this helper with 
`bool IsCopy = ZeCommandList == CommandBuffer->ZeCopyCommandList;`

Reply:
Indeed, we can do that.